tools/wds2idx.py

```diff
@@ -111,14 +111,13 @@ def _get_data_tar(self):
             if entry_type != b"-":
                 continue
 
-            offset = int(
-                next_blocks_line[next_blocks_line.find(b"block") + 6 : next_blocks_line.find(b":")]) * 512  # noqa: E203, E501
+            offset = int(next_blocks_line[next_blocks_line.find(b"block") + 6:
+                                          next_blocks_line.find(b":")])
+            offset = (offset + 1) * 512
 
             size = types_sizes_line[: -len(name)]
             size = size[: size.rfind(b"-") - 8]  # "... <size> 20yy-mm-...."
             size = int(size[size.rfind(b" "):])
-            offset -= size
-            offset -= offset % 512
```

Comment:
I am not really sure what is calculated here and what is the actual layout? Are the metadata at the end, that we can backtrack by the size and lower it to the closes multiple of 512?

Reply:
Added a comment.